docs/DISTRIBUTION-ARCHITECTURE.md

```
@@ -0,0 +1,260 @@
+    SRC -.->|"special case"| AGENTS
+    AGENTS -.-> PLUGIN
+
+    classDef src fill:#1f2937,color:#fff,stroke:#111;
+    classDef note fill:#fef3c7,color:#78350f,stroke:#d97706;
+    class SRC src;
+    class AGENTS note;
+```
+
+Plain-text view (terminals, diffs, non-Mermaid viewers):
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)